main.tf

```diff
@@ -106,6 +106,22 @@ resource "aws_s3_bucket" "default" {
       }
     }
   }
+
+  lifecycle {
+    ignore_changes = [
+      server_side_encryption_configuration
```

Comment:
just clarification: why ignore changes here?

Reply:
The AWS 3.x provider has two ways of enabling encryption:
- a deprecated flag on the `aws_s3_bucket`
- using `aws_s3_bucket_server_side_encryption_configuration`

When using the new `aws_s3_bucket_server_side_encryption_configuration` resource you need to add an ignore lifecycle rule on the s3_bucket to prevent it from trying to change the setting. This is not an issue in 4.x

Documentation: https://registry.terraform.io/providers/hashicorp/aws/3.75.1/docs/resources/s3_bucket_server_side_encryption_configuration#usage-notes